### PR TITLE
storage: use seek_for_prev

### PR DESCRIPTION
**Pull request by BusyJay**
Now that rocksdb supports `seek_for_prev`, we can use the API directly. Ref #1416.

@siddontang @disksing PTAL

### CODE REVIEW

File: src/storage/engine/mod.rs [hunk 1]
```diff
@@ -605,46 +597,65 @@ mod tests {
         })
     }
 
+    #[derive(PartialEq, Eq, Clone, Copy)]
+    enum SeekMode {
+        Normal,
+        Reverse,
```
Comment (siddontang) [inline src/storage/engine/mod.rs]: do we need Reverse now?

Comment (BusyJay) [inline src/storage/engine/mod.rs]: `near_reverse_seek` is still using. I prefer keeping `reverse_seek` in case it's needed again in the future.

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/raftstore/coprocessor/region_snapshot.rs [hunk 2]
```diff
@@ -234,6 +234,16 @@ impl<'a> RegionIterator<'a> {
         Ok(self.update_valid(true))
     }
 
+    pub fn seek_for_prev(&mut self, key: &[u8]) -> Result<bool> {
+        try!(self.should_seekable(key));
+        let key = keys::data_key(key);
+        self.valid = self.iter.seek_for_prev(key.as_slice().into());
+        if self.valid && self.iter.key() == self.end_key.as_slice() {
```
Comment (zhangjinpeng1987) [inline src/raftstore/coprocessor/region_snapshot.rs]: When this will happen?

**Comment by ngaut** [comment]
Please fix CI

File: src/raftstore/coprocessor/region_snapshot.rs (hunk 2 above)
Comment (BusyJay) [inline src/raftstore/coprocessor/region_snapshot.rs]: `seek_for_prev(self.end_key)`

Comment (zhangjinpeng1987) [inline src/raftstore/coprocessor/region_snapshot.rs]: This will return in L238, isn't it?

Comment (BusyJay) [inline src/raftstore/coprocessor/region_snapshot.rs]: No.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**